app/controllers/users_controller.rb

```diff
@@ -13,16 +21,66 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t :welcome
+      log_in @user
+      flash[:success] = t ".welcome"
       redirect_to @user
     else
       render :new
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    end
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    else
+      if @user.update_attributes user_params
+        flash[:success] = t ".profile_updated"
+        redirect_to @user
+      else
+        render :edit
+      end
+    end
+  end
+
+  def destroy
+    user = User.find_by(id: params[:id])
+    unless user.nil?
+      user.destroy
+      flash[:success] = t ".user_deleted"
+    else
+      flash[:warning] = t ".user_not_exist"
+    end
+    redirect_to users_url
```

Comment:
```
@user = User.find_by id: params[:id]

if @user
  # do something
else 
 # do another thing
end 
```
Now you can see that, the same logic is duplicated for 4 actions show, edit, update and destroy. Can you use before_action callback to remove this duplicate code.